lib/devise_saml_authenticatable/strategy.rb

```diff
@@ -57,6 +55,18 @@ def failed_auth(msg)
         Devise.saml_failed_callback.new.handle(@response, self) if Devise.saml_failed_callback
       end
 
+      def response_options
+        options = {
+          settings: saml_config(get_idp_entity_id(params)),
+          allowed_clock_drift: Devise.allowed_clock_drift_in_seconds,
+        }
+
+        if Devise.saml_validate_in_response_to
+          options[:matches_request_id] = request.session[:saml_transaction_id] || ""
```

Comment:
If the session is missing a `saml_transaction_id`, we'll send a blank string along to `ruby-saml`. That's because if we send `nil`, `ruby-saml` will skip the validation. Open to other suggestions if you have something that's more intuitive!

Reply:
Fair! What do you think about something like `"ID_MISSING"`?

Reply:
Sounds fine!

Reply:
Great! I handled that in https://github.com/apokalipto/devise_saml_authenticatable/pull/195/commits/0ebcdcc6a12d252cb541fdb1b636a4fdf01593f0.